vm/actor/tests/paych_actor_test.rs

```diff
@@ -0,0 +1,1265 @@
+    }
+
+    #[test]
+    fn actor_collect() {
+        struct TestCase {
+            dont_settle: bool,
+            exp_send_from: ExitCode,
+            exp_send_to: ExitCode,
+            exp_send_collect: ExitCode,
+        }
+
+        let test_cases = vec![
+            TestCase {
+                dont_settle: true,
+                exp_send_from: ExitCode::Ok,
+                exp_send_to: ExitCode::Ok,
+                exp_send_collect: ExitCode::ErrForbidden,
+            },
+            TestCase {
+                dont_settle: false,
+                exp_send_from: ExitCode::ErrPlaceholder,
+                exp_send_to: ExitCode::Ok,
+                exp_send_collect: ExitCode::ErrPlaceholder,
+            },
+            TestCase {
+                dont_settle: false,
+                exp_send_from: ExitCode::Ok,
+                exp_send_to: ExitCode::ErrPlaceholder,
+                exp_send_collect: ExitCode::ErrPlaceholder,
+            },
+        ];
+
+        for tc in test_cases {
+            let bs = MemoryDB::default();
+            let (mut rt, _, _sv) = require_create_cannel_with_lanes(&bs, 1);
+            rt.epoch = 10;
+            let mut state: PState = rt.get_state().unwrap();
+
+            if !tc.dont_settle {
+                rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+                rt.expect_validate_caller_addr(&[state.from, state.to]);
+                assert!(rt
+                    .call(
+                        &*PAYCH_ACTOR_CODE_ID,
+                        Method::Settle as u64,
+                        &Serialized::default()
+                    )
+                    .is_ok());
+                state = rt.get_state().unwrap();
+                assert_eq!(state.settling_at, 11);
+            }
+
+            // "wait" for SettlingAt epoch
+            rt.epoch = 12;
+
+            let sent_to_from = &rt.balance - state.to_send.clone();
+            rt.expect_send(
+                state.from,
+                METHOD_SEND,
+                Serialized::default(),
+                sent_to_from,
+                Serialized::default(),
+                tc.exp_send_from,
+            );
+            rt.expect_send(
+                state.to,
+                METHOD_SEND,
+                Serialized::default(),
+                state.to_send,
+                Serialized::default(),
+                tc.exp_send_to,
+            );
+
+            // Collect.
+            rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+            rt.expect_validate_caller_addr(&[state.from, state.to]);
+            let error = rt
+                .call(
+                    &*PAYCH_ACTOR_CODE_ID,
+                    Method::Collect as u64,
+                    &Serialized::default(),
+                )
+                .unwrap_err();
+            assert_eq!(tc.exp_send_collect, error.exit_code());
+        }
+    }
+}
+
+struct ActorHarness {
+    payer: Address,
+    payee: Address,
+}
```

Comment:
This isn't needed, there was no need to add this complexity to it as these are always just static values